bakery/service_test.go

```diff
@@ -131,10 +143,93 @@ func (s *ServiceSuite) TestMacaroonPaperFig6FailsWithBindingOnTamperedSignature(
 	}
 
 	// client makes request to ts.
-	err = ts.Check(d, strCompFirstPartyChecker(""))
+	err = ts.Check(d, strcmpChecker(""))
 	c.Assert(err, gc.ErrorMatches, "verification failed: signature mismatch after caveat verification")
 }
 
+func (s *ServiceSuite) TestNeedDeclared(c *gc.C) {
```

Comment:
I'd like a test case with a macaroon slice that fails to verify because a `needs-declared` declaration is missing from the discharges.

Wacky test cases I am curious about, just because it seems like they'd pass (and do we want them to?):
* Nested third-party caveats, which incrementally satisfy a `needs-declared`. Could be reasonable.
* Two third-party discharges that add `needs-declared` caveats, the discharges of which satisfy each other. Weird.
* A macaroon slice that adds a first-party caveat, with some caveats that are satisfied by a third-party `needs-declared` which it also adds.. the discharge for which does not duplicate any of the declarations. Weird and probably wrong.

Can you `needs-declared needs-declared needs-declared ...`?